Xamarin.Forms.Platform.UAP/ListViewRenderer.cs

```diff
@@ -171,7 +187,6 @@ protected override void OnElementPropertyChanged(object sender, PropertyChangedE
 			else if (e.PropertyName == ListView.ItemsSourceProperty.PropertyName)
 			{
 				ClearSizeEstimate();
-				ReloadData();
```

Comment:
Is there a use case I'm missing here for having this ReloadData call here?  The ReloadData is already called by the code inside *OnCollectionChanged*  so all this call was doing was causing the data to be initialized twice which was causing a COM exception

Reply:
@PureWeen This is absolutely needed! If you replace the entire collection attached to the Bindable Property this replaces the existing collection with a whole new collection/object reference.

CollectionChangedEvent only fires for changes within the collection, not if the object reference to the collection itself is changed. Once you've finished this PR, can you please retest against the broken repro project I reported here? https://github.com/xamarin/Xamarin.Forms/issues/2981 This will verify this behaviour.

Reply:
@mattrichnz   Thank you for the reference to the repro!! Been trying to get all those represented into the UI Test I have on this PR

So that repro was definitely still broken but for a different reason. The null check on *UpdateGrouping* was causing some basic things to not wire up 

While I agree with your assessment of how OnCollectionChanged works, the OnCollectionChanged that's part of ListViewRenderer doesn't correlate to the ObservableCollection that's set on the ListView.ItemSource. That OnCollectionChange is watching *TemplatedItemList* which forward CollectionChanged events to the ListViewRenderer

So when the entire ItemSource is swapped out the code here

https://github.com/xamarin/Xamarin.Forms/blob/a9a7277ad8d6b40a9f28081c248ba282a6476d95/Xamarin.Forms.Core/TemplatedItemsList.cs#L719
and here
https://github.com/xamarin/Xamarin.Forms/blob/a9a7277ad8d6b40a9f28081c248ba282a6476d95/Xamarin.Forms.Core/TemplatedItemsList.cs#L945

Propagates a Reset to ListViewRenderer which in turn calls *ReloadData*

I added your broken test case to this PR and for me it's now working after the latest changes
https://github.com/xamarin/Xamarin.Forms/pull/3023/files#diff-0edd99bdcd5cbb81e6610e90df74169aR125

Reply:
@PureWeen That's brilliant! We really appreciate all your hard work on this.